thirdparty/cmdconfig/commands/cmdeval/cmdeval.go

```diff
@@ -232,6 +296,8 @@ func (r *EvalFnRunner) preRunE(c *cobra.Command, args []string) error {
 		if err != nil {
 			return err
 		}
+	} else if r.AddImage {
+		return fmt.Errorf("--add requires --image")
```

Comment:
We shouldn't need this check. Only pre-requisite for `--save` is that function should evaluate successfully.

Reply:
ah, found the exec case in the https://kpt.dev/book/04-using-functions/01-declarative-function-execution